keps/0008-multi-cluster.yaml

```diff
@@ -33,20 +33,29 @@ Currently, KUTTL only supports a single test cluster. This KEP describes how we
 * Support applying resources across more than one test cluster.
 * Support asserting on resources across more than one test cluster.
 
-## Proposal
+## Proposals - TestStep API changes
+
+The proposal is to add a new setting within `TestStep`. We propose this field be called `runOnCluster` and of type `[]string`. This setting would allow the user to specify a list of the clusters set up by the `TestSuite` on which to run `TestSteps`.
+
+Recall that `TestSuite` contains a map of `KindConfig`s, entries in the `runOnCluster` would refer to one of the keys in this map. Where a user has specified homogenous clusters using `GlobalKindConfig` and `NumClusters`, a uniform naming scheme should exist to make identification of the right cluster to run on possible.
 
-The proposal is to add a new setting to the `TestStep` object: `kubeconfig`. This setting would allow the user to specify an alternative kubeconfig path to use for a given test step.
 
 ```
 apiVersion: kudo.dev/v1alpha1
 kind: TestStep
-kubeconfig: ./secondary-cluster.yaml
+runOnCluster:
+- my-cluster-1
+- my-cluster-2
 ```
 
-If the kubeconfig setting is not set, then the global Kubernetes client is used.
+If the runOnCluster setting is not set, then the `TestStep` will be run on all clusters. 
```

Comment:
it seems like we should have a configuration phase which sets all cluster to run on.  In this approach, it would be good if we see no defined clusters to mean what it means today.

Reply:
>  it would be good if we see no defined clusters to mean what it means today.

I'm hoping that this backwards compatibility is a consequence of what I'm proposing indeed. 

I was conceptualising the current arrangement as one where the tests are run on all clusters BUT there is actually only one cluster to run on. Which gives backwards compatibility if `runOnCluster` (or however we name that parameter) is unset.

Let me know if that makes sense.

> it seems like we should have a configuration phase which sets all cluster to run on. 

Yes, I think that this should happen per `TestStep` is the only thing, and be based on clusters declared via the `TestSuite` so that we can refer to a kubeconfig + kube context pair using strings.